pkg/start/probe.go

```diff
@@ -147,7 +147,33 @@ func (p *Probe) logEntry() *log.Entry {
 	return log.WithField("type", p.target.Type()).WithField("name", p.target.Name())
 }
 
-func (p *Probe) generateContent(ctx context.Context) (*blocks.BasicBlock, func(), error) {
+// Decide if content should be generated locally, and generate content
+func (p *Probe) getContent(ctx context.Context) (*cid.Cid, func(), error) {
+	switch p.target.(type) {
+	case ContentProvidingTarget:
+		return p.target.(ContentProvidingTarget).GenerateContent(ctx, p.config.PrivKey)
+	case Target:
+		return p.generateLocalContent(ctx)
+	}
+	return nil, nil, errors.New("Failed to generate content")
+}
```

Comment:
Why do we need this split? I think in both cases we'd generate random data locally. I could see that in case of the `UploadService` we wouldn't need to register it with the local blockstore, but that's actually not a problem. I may miss something here :)